jbpm-flow/src/main/java/org/jbpm/process/core/context/variable/VariableViolationException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.core.context.variable;
+
+/**
+ * Thrown when there is any kind of variable violation such as missing required variable
+ * or attempt to set already defined readonly variable. 
+ * 
+ */
+public class VariableViolationException extends RuntimeException {
+
+    private static final long serialVersionUID = 8031225233775014572L;
+
+    private final Long processInstanceId;
+    private final String variableName;
+    private final String errorMessage;
+
+    public VariableViolationException(Long processInstanceId, String variableName, String errorMessage) {
+        super("Variable '" + variableName + "' in process instance '" + (processInstanceId == null ? "unknown" : processInstanceId) + "' violated");
```

Comment:
Here it's overwriting the errorMessage passed by the caller.
```suggestion
        super(errorMessage);
```

Currently, it's displaying for case scenarios:

`Variable 'caseFile_s' in process instance '1' violated`

instead of the errorMessage tailored for variables starting with CASE_FILE_PREFIX: 

`Case file item 'caseFile_s' is required but not set`

If the process instance id is useful to be shown, then it may come from the caller itself.

Reply:
that's the whole point, the message that comes from parent is generic that indicates about violation, while it has extra errorMessage property that says the specific reason for it.

Reply:
ok, my confusion was because in the stacktrace I only see the parent (generic) message